backend/graphql/index.ts

```diff
@@ -43,6 +43,16 @@ const isValidDate = (dateString: string) => {
   );
 };
 
+const isValidDateTime = (dateTimeString: string) => {
+  const regEx = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}$/; // YYYY-MM-DDTHH:mm
+  if (!dateTimeString.match(regEx)) return false; // Invalid format
+  return (
+    !Number.isNaN(Date.parse(`${dateTimeString}:00`)) && // cover cases of DD > 31
+    new Date(`${dateTimeString}:00`).toISOString().slice(0, 16) ===
```

Comment:
I think we actually need to specify the timezone as UTC for this comparison to work, otherwise the ```new Date(`${dateTimeString}:00`)``` will be interpreted in the server's local time which may not match the original datetime string.

```suggestion
    new Date(`${dateTimeString}:00+00:00`).toISOString().slice(0, 16) ===
```